quantus/metrics/faithfulness/sufficiency.py

```diff
@@ -295,39 +278,11 @@ def evaluate_instance(
             return 0
         return np.sum(pred_low_dist_a == pred_a) / len(low_dist_a)
 
-    def custom_preprocess(
-        self,
-        model: ModelInterface,
-        x_batch: np.ndarray,
-        y_batch: Optional[np.ndarray],
-        a_batch: Optional[np.ndarray],
-        s_batch: np.ndarray,
-        custom_batch: Optional[np.ndarray],
-    ) -> Dict[str, Any]:
-        """
-        Implementation of custom_preprocess_batch.
-
-        Parameters
-        ----------
-        model: torch.nn.Module, tf.keras.Model
-            A torch or tensorflow model e.g., torchvision.models that is subject to explanation.
-        x_batch: np.ndarray
-            A np.ndarray which contains the input data that are explained.
-        y_batch: np.ndarray
-            A np.ndarray which contains the output labels that are explained.
-        a_batch: np.ndarray, optional
-            A np.ndarray which contains pre-computed attributions i.e., explanations.
-        s_batch: np.ndarray, optional
-            A np.ndarray which contains segmentation masks that matches the input.
-        custom_batch: any
-            Gives flexibility ot the user to use for evaluation, can hold any variable.
-
-        Returns
-        -------
-        dictionary[str, np.ndarray]
-            Output dictionary with 'a_sim_vector_batch' as and attributtion similarity matrix as value.
-        """
-
+    def batch_preprocess(self, data_batch: Dict[str, Any]) -> Dict[str, Any]:
+        data_batch = super().batch_preprocess(data_batch)
+        model = data_batch["model"]
+        x_batch = data_batch["x_batch"]
+        a_batch = data_batch["a_batch"]
         a_batch_flat = a_batch.reshape(a_batch.shape[0], -1)
```

Comment:
please add comments, I recogise this from prev implementation but please add some comments